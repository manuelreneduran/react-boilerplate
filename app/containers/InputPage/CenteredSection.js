import styled from 'styled-components';

const CenteredSection = styled.section`
  display: flex;
  flex-direction: row;
  justify-content: center;
`;

export default CenteredSection;
