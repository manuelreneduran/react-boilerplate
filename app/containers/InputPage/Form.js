import styled from 'styled-components';

const Form = styled.form`
  min-height: 30vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
`;

export default Form;
