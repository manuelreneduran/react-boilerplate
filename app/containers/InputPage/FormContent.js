import styled from 'styled-components';

const FormContent = styled.div`
  margin: 1em 0;
  min-height: 6vh;
`;

export default FormContent;
