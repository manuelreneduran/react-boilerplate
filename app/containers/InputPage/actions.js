import {
  CHANGE_STRING_INPUT,
  ADD_STRING,
  ADD_STRING_SUCCESS,
  ADD_STRING_FAILURE,
} from './constants';

/**
 * Changes the input field of the form
 *
 * @param  {string} stringInput The new text of the input field
 *
 * @return {object} An action object with a type of CHANGE_STRING_INPUT
 */
export function changeStringInput(stringInput) {
  return {
    type: CHANGE_STRING_INPUT,
    stringInput,
  };
}

/**
 * Add a string to DB, this action starts request saga
 *
 * @return {object}     An action object with a type of ADD_STRING
 */
export function addString() {
  return {
    type: ADD_STRING,
  };
}

/**
 * Dispatched when the string is added to db by the request saga
 *
 * @return {object}      An action object with a type of ADD_STRING_SUCCESS passing the repos
 */
export function addStringSuccess() {
  return {
    type: ADD_STRING_SUCCESS,
  };
}

/**
 * Dispatched when adding a string to DB fails
 *
 * @param  {object} error The error
 *
 * @return {object}       An action object with a type of ADD_STRING_ERROR passing the error
 */
export function addStringFailure(error) {
  return {
    type: ADD_STRING_FAILURE,
    error,
  };
}
