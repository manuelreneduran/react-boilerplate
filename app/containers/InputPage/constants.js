export const CHANGE_STRING_INPUT = 'app/InputPage/CHANGE_STRING_INPUT';
export const ADD_STRING = 'app/InputPage/ADD_STRING';
export const ADD_STRING_SUCCESS = 'app/InputPage/ADD_STRING_SUCCESS';
export const ADD_STRING_FAILURE = 'app/InputPage/ADD_STRING_FAILURE';
