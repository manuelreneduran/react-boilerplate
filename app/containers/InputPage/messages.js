/*
 * Input Page Messages
 *
 * This contains all the text for the InputPage container.
 */
import { defineMessages } from 'react-intl';

export const scope = 'app.containers.InputPage';

export default defineMessages({
  header: {
    id: `${scope}.header`,
    defaultMessage: 'This is the Input Page',
  },
  pageInfoMessage: {
    id: `${scope}.pageInfoMessage`,
    defaultMessage: 'Write something and send it to the server!',
  },
});
