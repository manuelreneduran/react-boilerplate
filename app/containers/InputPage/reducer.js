import produce from 'immer';
import {
  CHANGE_STRING_INPUT,
  ADD_STRING_SUCCESS,
  ADD_STRING_FAILURE,
  ADD_STRING,
} from './constants';

// The initial state of the App
export const initialState = {
  loading: false,
  error: false,
  complete: false,
  data: {
    stringInput: '',
  },
};

/* eslint-disable default-case, no-param-reassign */
const inputReducer = (state = initialState, action) =>
  produce(state, draft => {
    switch (action.type) {
      case CHANGE_STRING_INPUT:
        // Delete prefixed '@' from the github username
        draft.data.stringInput = action.stringInput;
        draft.complete = false;
        draft.error = false;
        break;
      case ADD_STRING:
        draft.loading = true;
        draft.error = false;
        draft.complete = false;
        break;
      case ADD_STRING_SUCCESS:
        draft.complete = true;
        draft.loading = false;
        draft.error = false;
        draft.data.stringInput = '';
        break;
      case ADD_STRING_FAILURE:
        draft.complete = true;
        draft.loading = false;
        draft.error = action.error;
        break;
    }
  });

export default inputReducer;
