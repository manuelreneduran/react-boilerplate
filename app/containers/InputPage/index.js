/*
 * InputPage
 *
 * This page allows users to input a string and send to server
 *
 */

import React, { memo } from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { FormattedMessage } from 'react-intl';
import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import {
  makeSelectInput,
  makeSelectError,
  makeSelectLoading,
  makeSelectComplete,
} from './selectors';
import InputLabel from '../../components/InputLabel';
import { useInjectReducer } from 'utils/injectReducer';
import { useInjectSaga } from 'utils/injectSaga';
import { changeStringInput, addString } from './actions';
import reducer from './reducer';
import messages from './messages';
import saga from './saga';

const key = 'inputPage';

function InputPage({
  stringInput,
  onInputChange,
  onSubmitForm,
  error,
  loading,
  complete,
}) {
  useInjectReducer({ key, reducer });
  useInjectSaga({ key, saga });

  const inputLabelProps = {
    loading,
    error,
  };
  return (
    <div>
      <Helmet>
        <title>Input Page</title>
        <meta name="description" content="An input page" />
      </Helmet>
      <h1>
        <FormattedMessage {...messages.header} />
      </h1>
      <h3>
        <FormattedMessage {...messages.pageInfoMessage} />
      </h3>
      <form onSubmit={onSubmitForm}>
        <input
          onChange={onInputChange}
          value={stringInput}
          type="text"
          placeholder="Add a string name"
        />
        <button type="submit">Add string</button>
        {complete && <InputLabel {...inputLabelProps} />}
      </form>
    </div>
  );
}

InputPage.propTypes = {
  stringInput: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  stringInput: makeSelectInput(),
  error: makeSelectError(),
  loading: makeSelectLoading(),
  complete: makeSelectComplete(),
});

export function mapDispatchToProps(dispatch) {
  return {
    onInputChange: evt => {
      if (evt !== undefined && evt.preventDefault) evt.preventDefault();
      dispatch(changeStringInput(evt.target.value));
    },
    onSubmitForm: evt => {
      if (evt !== undefined && evt.preventDefault) evt.preventDefault();
      dispatch(addString());
    },
  };
}

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

export default compose(
  withConnect,
  memo,
)(InputPage);
