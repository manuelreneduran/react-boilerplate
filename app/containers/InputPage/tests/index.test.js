/**
 * Test the InputPage
 */

import React from 'react';
import { render, fireEvent, waitForElement } from 'react-testing-library';
import { IntlProvider } from 'react-intl';
import { Provider } from 'react-redux';
import { browserHistory } from 'react-router-dom';

import InputPage from '../index';
import configureStore from '../../../configureStore';

const defaultProps = {
  stringInput: '',
  onInputChange: jest.fn(),
  onSubmitForm: jest.fn(),
  error: false,
  loading: false,
  complete: false,
  setErrors: jest.fn(),
};

describe('<InputPage />', () => {
  let store;

  beforeAll(() => {
    store = configureStore({}, browserHistory);
  });

  it('should render and match the snapshot', () => {
    // const component = setup();
    const {
      container: { firstChild },
    } = render(
      <Provider store={store}>
        <IntlProvider locale="en">
          <InputPage {...defaultProps} />
        </IntlProvider>
      </Provider>,
    );
    expect(firstChild).toMatchSnapshot();
  });

  describe('Events', () => {
    it('should show a warning message if submitted input text is empty', async () => {
      const { getByTestId } = render(
        <Provider store={store}>
          <IntlProvider locale="en">
            <InputPage stringInput="blahaha" />
          </IntlProvider>
        </Provider>,
      );
      fireEvent.submit(getByTestId('form'));
      const label = await waitForElement(() => getByTestId('string-input'));
      expect(label).toHaveTextContent('You must have at least one character');
    });
  });
});
