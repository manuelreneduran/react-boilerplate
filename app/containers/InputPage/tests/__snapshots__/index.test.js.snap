// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputPage /> should render and match the snapshot 1`] = `
<div>
  <h1
    class="H1-tncdg6-0 cJREqO"
  >
    <span>
      This is the Input Page
    </span>
  </h1>
  <h3
    class="H3-sc-1nwhpi5-0 heZsqC"
  >
    <span>
      Write something and send it to the server!
    </span>
  </h3>
  <section
    class="CenteredSection-sc-1jg9oos-0 gJqWTy"
  >
    <form
      class="Form-sc-4e0erd-0 gGPaDr"
      data-testid="form"
    >
      <div
        class="FormContent-sc-1k2sc3h-0 gCaPjC"
      >
        <input
          class="Input-sc-19txxuq-0 hjivrB"
          placeholder="Add a string name"
          type="text"
          value=""
        />
      </div>
      <div
        class="FormContent-sc-1k2sc3h-0 gCaPjC"
      >
        <button
          role="button"
          type="submit"
        >
          Add string
        </button>
      </div>
      <div
        class="FormContent-sc-1k2sc3h-0 gCaPjC"
      >
         
         
      </div>
    </form>
  </section>
</div>
`;
